### PR TITLE
Fix the azstackhcivm update bug

### PR DESCRIPTION
**Pull request by shraddhasun**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [x] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/StackHCIVM/StackHCIVM.Autorest/test/AzStackHCIVMImageNew.Tests.ps1 [hunk 1]
```diff
@@ -15,13 +15,9 @@ if(($null -eq $TestName) -or ($TestName -contains 'AzStackHCIVMImageNew'))
 }
 
 Describe 'AzStackHCIVMImageNew' {
-    It 'Create Image '  {
```
Comment (dolauli) [inline src/StackHCIVM/StackHCIVM.Autorest/test/AzStackHCIVMImageNew.Tests.ps1:18]: The modifications made in the "test" section appear unrelated to the fix. Please include a test specifically addressing the issue fix in the "Add-AzStackHCIVMVirtualMachineDataDisk" function and remove unneeded test changes.

**Review by dolauli — CHANGES_REQUESTED** [review]

**Comment by shraddhasun** [comment]
\azp run

File: src/StackHCIVM/StackHCIVM.Autorest/test/AzStackHCIVMImageNew.Tests.ps1 (hunk 1 above)
Comment (shraddhasun) [reply src/StackHCIVM/StackHCIVM.Autorest/test/AzStackHCIVMImageNew.Tests.ps1:18]: Undid this. Most of the test changes are related to modifying names because the old rg doesnt exist anymore.

File: src/StackHCIVM/StackHCIVM.Autorest/test/AzStackHCIVMVirtualMachine.Tests.ps1 [hunk 2]
```diff
@@ -31,23 +31,85 @@ Describe 'AzStackHCIVMVirtualMachine' {
...
+
+    It 'Create Network Interface  '  {
+        New-AzStackHciVMNetworkInterface  -Name "testNic1" -SubscriptionId $env.subscriptionId -ResourceGroupName $env.resourceGroupName -CustomLocationId $env.customLocationId -Location $env.location -SubnetId "/subscriptions/37908b1f-2848-4c85-b8bf-a2cab2c3b0ba/resourceGroups/vpclus0724-rg/providers/Microsoft.AzureStackHCI/logicalnetworks/lnet_aug"  | Select-Object -Property ProvisioningState | Should -BeExactly "@{ProvisioningState=Succeeded}"
     }
 
 
-    It 'Delete'{
+    It 'Add NIC'  {
         {
+            Add-AzStackHCIVMVirtualMachineNetworkInterface -Name manualvmtest2 -ResourceGroupName $env.resourceGroupName -NicName "testNic1"
```
Comment (shraddhasun) [inline src/StackHCIVM/StackHCIVM.Autorest/test/AzStackHCIVMVirtualMachine.Tests.ps1:62]: Added the vm update tests here

**Comment by msJinLei** [comment]
```
    [-] Create vm with osdisk   45ms
      WriteErrorException: A Network Interface with id: /subscriptions/37908b1f-2848-4c85-b8bf-a2cab2c3b0ba/resourceGroups/vpclus0724-rg/providers/Microsoft.AzureStackHCI/networkinterfaces/testNic2 does not exist.
      at New-AzStackHCIVMVirtualMachine, D:\a\_work\1\s\artifacts\Debug\Az.StackHCIVM\StackHCIVM.Autorest\custom\New-AzStackHCIVMVirtualMachine_New.ps1: line 524
      at New-AzStackHCIVMVirtualMachine<End>, D:\a\_work\1\s\artifacts\Debug\Az.StackHCIVM\StackHCIVM.Autorest\exports\ProxyCmdletDefinitions.ps1: line 4338
      at <ScriptBlock>, D:\a\_work\1\s\artifacts\Debug\Az.StackHCIVM\StackHCIVM.Autorest\test\AzStackHCIVMVirtualMachine.Tests.ps1: line 77
    [+] Add Data Disk 0ms
WARNING: Running this command will delete the virtual machine.
    [+] Delete 60.21s
WARNING: Running this command will delete the virtual hard disk.
    [-] Delete resources 57ms
      UndeclaredResponseException: admission webhook "deletionvalidationwebhook.infrastructure.azstackhci.microsoft.com" denied the request: Deletion of default/testvhddisk11-b1347c81-d1fe-46c2-a4c9-bd3a7f466bb5 failed. It is in use by other resources. Resource type: *v1alpha1.AzStackHCIVirtualHardDisk
      at Remove-AzStackHCIVMVirtualHardDisk<Process>, D:\a\_work\1\s\artifacts\Debug\Az.StackHCIVM\StackHCIVM.Autorest\internal\ProxyCmdletDefinitions.ps1: line 6005
      at Remove-AzStackHCIVMVirtualHardDisk<Process>, D:\a\_work\1\s\artifacts\Debug\Az.StackHCIVM\StackHCIVM.Autorest\custom\Remove-AzStackHCIVmVirtualHardDisk_New.ps1: line 142
      at Remove-AzStackHCIVMVirtualHardDisk<Process>, D:\a\_work\1\s\artifacts\Debug\Az.StackHCIVM\StackHCIVM.Autorest\exports\ProxyCmdletDefinitions.ps1: line 5299
      at <ScriptBlock>, D:\a\_work\1\s\artifacts\Debug\Az.StackHCIVM\StackHCIVM.Autorest\test\AzStackHCIVMVirtualMachine.Tests.ps1: line 105

```

**Comment by shraddhasun** [comment]
\azp run

File: src/StackHCIVM/StackHCIVM.Autorest/custom/New-AzStackHCIVMVirtualMachine_New.ps1 [hunk 3]
```diff
@@ -551,7 +551,7 @@ function New-AzStackHCIVMVirtualMachine {
     }
     if (-not $ProvisionVMConfigAgent){
       $null = $PSBoundParameters.Remove("WindowConfigurationProvisionVMConfigAgent")
-      $PSBoundParameters.Add("WindowConfigurationProvisionVMConfigAgent", $true)
+      $PSBoundParameters.Add("WindowConfigurationProvisionVMConfigAgent", $false)
     }
```
Comment (dolauli) [inline src/StackHCIVM/StackHCIVM.Autorest/custom/New-AzStackHCIVMVirtualMachine_New.ps1:555]: Does this change fix another issue?

File: src/StackHCIVM/StackHCIVM.Autorest/docs/Get-AzStackHCIVMImage.md [hunk 4]
```diff
@@ -48,7 +48,7 @@ Get-AzStackHCIVMImage -Name "testimage" -ResourceGroupName "test-rg"
 ```output
 Name            ResourceGroupName
 ----            -----------------
-testImage       test-rg
```
Comment (dolauli) [inline src/StackHCIVM/StackHCIVM.Autorest/docs/Get-AzStackHCIVMImage.md:51]: Why do you make this change. The original one seems better.

**Review by dolauli — CHANGES_REQUESTED** [review]

File: src/StackHCIVM/StackHCIVM.Autorest/custom/New-AzStackHCIVMVirtualMachine_New.ps1 (hunk 3 above)
Comment (shraddhasun) [reply src/StackHCIVM/StackHCIVM.Autorest/custom/New-AzStackHCIVMVirtualMachine_New.ps1:555]: yes, it does. This was incorrect.

File: src/StackHCIVM/StackHCIVM.Autorest/docs/Get-AzStackHCIVMImage.md (hunk 4 above)
Comment (shraddhasun) [reply src/StackHCIVM/StackHCIVM.Autorest/docs/Get-AzStackHCIVMImage.md:51]: updated

**Review by dolauli — APPROVED** [review]
